packages/ts/react-crud/src/autogrid.tsx

```diff
@@ -258,6 +265,17 @@ function useColumns(
 
   columns = addCustomColumns(columns, options, setColumnFilter);
 
+  // When using `hiddenColumns` option, remove columns to hide using their `key`
+  if (options.hiddenColumns) {
+    columns = columns.filter((column) => {
+      const { key } = column;
+      if (key) {
+        return !options.hiddenColumns?.includes(key);
+      }
+      return true;
+    });
```

Comment:
Line 275 is not covered by tests. Now, the case of a null key is not a real one, since all columns are supposed to have one. Yet, that makes me think of a shorter syntax for this filter:
```suggestion
    columns = columns.filter(({ key }) => !(key && options.hiddenColumns?.includes(key)));
```

Reply:
I also noticed the missing test coverage too and added an additional test `should ignore unknown columns when using hiddenColumns` with https://github.com/vaadin/hilla/compare/ce783d2e6ad87a7c921edfe733c9d89ca6396369..e89d85722221ca1641fb841447a1f4dd9830bb85. Could you rerun the code coverage test? The codecov report is based on an older commit.

Reply:
The report seems to be up-to-date. The `key` is related to `columns` here, while your test adds unknown column names to `hiddenColumns`. But, again, all columns are supposed to have a key, even custom ones, see https://github.com/rbrki07/hilla/blob/376e8f2017ae60865b8561cf4b61e26633d9a9a6/packages/ts/react-crud/test/autogrid.spec.tsx#L1310 for example.

Reply:
Thank you for clarification on this. I updated the pr with you suggestion.